.github/workflows/release_trial_frontend.yml

```diff
@@ -2,42 +2,70 @@
 name: Release Trial Frontend
 
 on:
-  push:
-    branches:
-      - trialfrontend01
-      - trialfrontend02
-      - trialfrontend03
-
-defaults:
-  run:
-    working-directory: prime-router
+  pull_request:
+    types:
+      - labeled
 
 jobs:
   pre_job:
     name: "Set Build Environment"
     runs-on: ubuntu-latest
     outputs:
-      env_name: ${{ steps.build_vars.outputs.env_name }}
-      version: ${{ github.ref_name }}
+      # env_name and version are the same for trial frontends
+      env_name: ${{ steps.set_vars.outputs.env_name }}
+      version: ${{ steps.set_vars.outputs.version }}
+      trialfrontend_url: ${{ steps.set_vars.outputs.trialfrontend_url }}
     steps:
-      - name: Check out changes
-        uses: actions/checkout@8f4b7f84864484a7bf31766abe9204da3cbe65b3
-
-      - name: Build vars
-        id: build_vars
-        uses: ./.github/actions/build-vars
+      - name: Set vars
+        id: set_vars
+        if: |
+          (github.event.label.name == 'trialfrontend01') || 
+          (github.event.label.name == 'trialfrontend02') || 
+          (github.event.label.name == 'trialfrontend03')
+        shell: bash
+        run: |
+          echo "env_name=${{ github.event.label.name }}" >> $GITHUB_OUTPUT
+          echo "version=${{ github.event.label.name }}" >> $GITHUB_OUTPUT
+          if [[ '${{ github.event.label.name }}' == 'trialfrontend01' ]]; then
+            echo "trialfrontend_url=https://pdhstagingpublictrial01.z13.web.core.windows.net" >> $GITHUB_OUTPUT
+          elif [[ '${{ github.event.label.name }}' == 'trialfrontend02' ]]; then
+            echo "trialfrontend_url=https://pdhstagingpublictrial02.z13.web.core.windows.net" >> $GITHUB_OUTPUT
+          elif [[ '${{ github.event.label.name }}' == 'trialfrontend03' ]]; then
+            echo "trialfrontend_url=https://pdhstagingpublictrial03.z13.web.core.windows.net" >> $GITHUB_OUTPUT
+          fi
 
   build_frontend_react_release:
     name: "Release: Build Frontend (React)"
     needs:
       - pre_job
+    if: ${{ needs.pre_job.outputs.trialfrontend_url }} != ""
     runs-on: ubuntu-latest
     defaults:
       run:
         working-directory: frontend-react
     steps:
+      - name: Find Comment
+        id: comment_find
+        uses: peter-evans/find-comment@v2
```

Comment:
We use [SHA hashes](https://devopsjournal.io/blog/2021/12/11/GitHub-Actions-Maturity-Levels) for all public actions

`peter-evans/find-comment@034abe94d3191f9c89d870519735beae326f2bdb`

Reply:
Got it -- thanks for the context.  Will update here and in the other pull request :)

Reply:
Updated!  Feel free to take another look whenever!